biobalm/interaction_graph_utils.py

```diff
@@ -181,3 +181,34 @@ def source_SCCs(bn: BooleanNetwork) -> list[list[str]]:
             result.append(scc_names)
 
     return sorted(result)
+
+
+def source_nodes(
+    network: BooleanNetwork, ctx: SymbolicContext | None = None
+) -> list[str]:
+    """
```

Comment:
Could you add the parameters and returns to this docstring?

Reply:
Or I suppose I could do it; I think it's straightforward here, and it looks like it's the only one that needs it.

Reply:
(I just did this)